### DISCUSSION TITLE
loadMembers occasionally returns 0 members

### DISCUSSION THREAD

**Discussion by yHSJ** [Chew/JDA-Chewtils#62] [Q&A] [answered]
I am running a bot that regularly (30 minutes) runs the guild#loadMembers() method and kicks specific members. Recently, I've noticed that every so often certain guilds will return 0 members after loadMembers runs and never loads the true member count again until I restart my bot. There are no exceptions thrown.

Is this an issue with rate limiting on discord's side? Or is there a situation that should be throwing an error and isn't?

**Comment by Chew** [comment, marked as answer]
This is probably a JDA issue so I can't help sadly :/

I always recommend caching all members anyway, so you can just getMembers.

**Reply by yHSJ** [reply]
Thanks so much for the quick response! I do cache all members, but I thought that caching was only on the bot starting, not continuous whenever a new user joins a guild. Am I wrong on that?

**Reply by Chew** [reply]
I believe it auto updates, I never have inconsistencies.

**Reply by Andre601** [reply]
How a bot caches members depends on certain factors. Namely:

- Does the bot have the `GUILD_MEMBERS` intent enabled? It's privileged so you have to enable it in your bot's code and on the Dashboard of Discord.
- Is caching set to actively cache all members? Default is `none` which means it only caches the self member. You would need to use the `ALL` caching policy. [[Wiki of JDA](https://github.com/DV8FromTheWorld/JDA/wiki/Gateway-Intents-and-Member-Cache-Policy)]

Personally speaking, do I recommend not caching all members, as this will significantly increase the RAM and CPU usage of your bot the more members and servers it's on.
You should work with events as much as possible to keep the cache low and still handle members.
